src/triage/component/architect/state_table_generators.py

```diff
@@ -151,33 +151,25 @@ def _create_and_populate_sparse_table(self, as_of_dates):
         """
 
         self.db_engine.execute(
-            '''create table {sparse_state_table} (
+            f'''create table {self.sparse_table_name} (
                 entity_id integer,
                 as_of_date timestamp,
-                {active_state} boolean
-            )
-            '''.format(
-                sparse_state_table=self.sparse_table_name,
-                active_state=DEFAULT_ACTIVE_STATE
+                {DEFAULT_ACTIVE_STATE} boolean
             )
+            '''
         )
         logging.info('Created sparse state table, now inserting rows')
+
         for as_of_date in as_of_dates:
-            formatted_date = "'{}'::timestamp".format(as_of_date.isoformat())
+            formatted_date = f"{as_of_date.isoformat()}"
             dated_query = self.query.replace('{as_of_date}', formatted_date)
-            full_query = '''insert into {sparse_state_table}
-                select q.entity_id, {as_of_date}, true
-                from ({query}) q
+            full_query = f'''insert into {self.sparse_table_name}
+                select q.entity_id, '{formatted_date}'::timestamp, true
```

Comment:
I don't think I understand – didn't you want to remove `::timestamp` casting from our queries, and have the user configuration do this instead? I think I just don't see what's been functionally changed here.

Reply:
This particular query is not a user-configurable query